### PR TITLE
fix(components/packages): standardize how schematics visit project files

### PR DESCRIPTION
**Pull request by johnhwhite**
[AB#2812991](https://dev.azure.com/blackbaud/f565481a-7bc9-4083-95d5-4f953da6d499/_workitems/edit/2812991)

### CODE REVIEW

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/1967/storybook/)

Component Storybooks:

- _(no component storybooks affected in this pr)_ 

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/1967/code-examples/)
- [dep-graph](https://blackbaud.github.io/skyux-pr-preview/1967/dep-graph/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/1967/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/1967/playground/)

File: libs/components/packages/src/schematics/utility/visit-project-files.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { Tree } from '@angular-devkit/schematics';
+import { FileVisitor } from '@angular-devkit/schematics/src/tree/interface';
+
+const rootIgnore = ['dist', 'coverage'];
+const alwaysIgnore = ['node_modules'];
```
Comment (Blackbaud-SteveBrush) [inline libs/components/packages/src/schematics/utility/visit-project-files.ts:5]: What's the difference between "root" ignore and "always" ignore?

File: libs/components/packages/src/schematics/utility/visit-project-files.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+export function visitProjectFiles(
+  tree: Tree,
+  projectPath: string,
+  visitor: FileVisitor,
+): void {
+  tree.getDir(projectPath).visit((path, entry) => {
+    if (
+      path.includes('/.') ||
```
Comment (Blackbaud-SteveBrush) [inline libs/components/packages/src/schematics/utility/visit-project-files.ts:14]: Is this matching filenames that start with `.`? Or relative paths? e.g. `../../../foobar`

Comment (johnhwhite) [reply libs/components/packages/src/schematics/utility/visit-project-files.ts:14]: Filenames that start with `.`

File: libs/components/packages/src/schematics/utility/visit-project-files.ts (hunk 1 above)
Comment (johnhwhite) [reply libs/components/packages/src/schematics/utility/visit-project-files.ts:5]: It's possible to use `dist` or `coverage` as directories in a project (it'd be unfortunate but not prohibited), so only ignore them at the root of a workspace, whereas `node_modules` should be ignored at any level.

File: libs/components/packages/src/schematics/utility/visit-project-files.ts (hunk 2 above)
Comment (Blackbaud-SteveBrush) [reply libs/components/packages/src/schematics/utility/visit-project-files.ts:14]: Wouldn't `../../../foobar` also be matched with this change?

Comment (johnhwhite) [reply libs/components/packages/src/schematics/utility/visit-project-files.ts:14]: The `Tree` `visit` method only walks down the tree, so these wouldn't matter.

**Review by Blackbaud-SteveBrush — APPROVED** [review]

**Comment by blackbaud-sky-build-user** [comment]
Cherry-pick successful! :cherries: :tada: See https://github.com/blackbaud/skyux/pull/1970

**Pull request merged**